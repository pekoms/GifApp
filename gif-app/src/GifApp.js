import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import GifGrid from './components/GifGrid'


export const GifApp = ({defaultCategory=[]}) => {

    //const categories = ['One Punch','Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategory)
    /*const handleAdd = () => {
        setCategories([...categories, 'HunterXHunter']);
    }*/
    return (
        <div>
            <h2>GifApp</h2>
            <AddCategory setCategories={setCategories}></AddCategory>
            <hr />

            <ul>
                {
                    categories.map(category => 
                        <GifGrid 
                            key={category}
                            category={category}></GifGrid>
                    )
                }
            </ul>
        </div>
    )
}
