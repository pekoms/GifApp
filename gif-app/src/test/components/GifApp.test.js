import { shallow } from "enzyme"
import { GifApp } from "../../GifApp"

let wrapper=shallow(<GifApp/>)
describe('Test de GifAppTest', () => {

    test('El snapshot cuadra',()=>{
        expect(wrapper).toMatchSnapshot();      
    })

    test('Debe mostrar una lista de catergorias',()=>{
        const categories =['Pokemon','Digimon'];
        wrapper=shallow(<GifApp defaultCategory={categories}/>)
        expect(wrapper).toMatchSnapshot();     
        expect(wrapper.find('GifGrid').length).toBe(categories.length);      


    })
    
})
