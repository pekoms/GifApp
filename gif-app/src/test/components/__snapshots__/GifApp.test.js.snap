// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifAppTest Debe mostrar una lista de catergorias 1`] = `
<div>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Pokemon"
      key="Pokemon"
    />
    <GifGrid
      category="Digimon"
      key="Digimon"
    />
  </ul>
</div>
`;

exports[`Test de GifAppTest El snapshot cuadra 1`] = `
<div>
  <h2>
    GifApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</div>
`;
